plugins/GSdx/GSTextureCache.h

```diff
@@ -105,6 +106,58 @@ class GSTextureCache
 		void Update();
 	};
 
+	class Palette
+	{
+	private:
+		uint16 m_ref_counter = 0; // Reference counter
+		uint32* m_clut; // Pointer to a copy of relevant clut
+		GSTexture* m_tex_palette; // Pointer to valid texture with relevant clut as content
+
+	public:
+		Palette(); // Default constructor
+		Palette(const Palette& palette); // Copy constructor
+		Palette(const GSRenderer* renderer, uint16 pal); // Creates a copy of the current clut and a texture with its content
+
+		// Getter for clut pointer
+		__forceinline uint32* GetClut();
+
+		// Getter for palette texture pointer
+		__forceinline GSTexture* GetPaletteGSTexture();
+
+		// Getter for the reference counter
+		__forceinline uint16 GetRefCounter();
+
+		// Increment the reference counter by 1
+		__forceinline void IncrementRefCounter();
+
+		// Decrement the reference counter by 1
+		__forceinline void DecrementRefCounter();
+
+		// Recycles palette texture and frees clut copy
+		__forceinline void Dispose(const GSRenderer* renderer);
```

Comment:
It was just to avoid keeping the reference to the renderer saved in each object, but I agree with the fact that it would enable doing the cleanup in the destructor, which is really nicer. Will refactor.